UndertaleModTool/Windows/FindReferencesTypesDialog/FindReferencesTypesDialog.xaml.cs

```diff
@@ -173,9 +187,19 @@ private async void SearchButton_Click(object sender, RoutedEventArgs e)
                 }
 
                 Hide();
-                var results = await UndertaleResourceReferenceMethodsMap.GetUnreferencedObjects(data, typesDict);
-                FindReferencesResults dialog = new(data, results);
-                dialog.Show();
+                FindReferencesResults dialog = null;
+                try
+                {
+                    var results = await UndertaleResourceReferenceMethodsMap.GetUnreferencedObjects(data, typesDict);
+                    dialog = new(data, results);
+                    dialog.Show();
+                }
+                catch (Exception ex)
```

Comment:
can we somehow trickle the exception type further instead of just having a catchall?

Reply:
As far as I understand we can't. This function is called directly from a button in XAML, so if we don't catch it here the application will crash.
This isn't a silent catchall and it should never be reached unless there is a bug. It is only here to stop the application from crashing if there is a bug, which is exactly what happens in #1946 and is way worse than the search just failing and showing an error message.

Reply:
Yeah, if it's just to prevent the entire application stability from being compromised, I'm OK with a catchall like this. Just need confirmation from @Miepee too

Reply:
well i meant more having a specicif exception type
so instead of `catch (Exception ex)` have `catch (InvalidAccessException ex)` or whatever exception type is used here.

Reply:
So in this case the crash was caused by a null pointer dereference (which also became an AggregateException), but since the point of it is to protect the user in the future, regardless of whatever bugs get introduced in the search process, I think plain Exception is probably the best. It's not like there's documented behavior for a bug we're not sure exists yet.

Reply:
eh, sure